libs/remote_storage/src/s3_bucket.rs

```diff
@@ -67,6 +68,9 @@ impl S3Bucket {
             aws_config.bucket_name
         );
 
+        let provider_conf = ProviderConfig::without_region()
+            .with_region(Some(Region::new(aws_config.bucket_region.clone())));
```

Comment:
Do I assume correctly that this needs the region because the tokens are scoped to region, or something else? Should we comment the reason?

we could move this into a new block defining the WebIdentityToken and returning it... Though the signature `WebIdentityTokenCR::builder().configure(&provider)` taking a borrow does seem weird, but it should work.

It would become clear that we don't need this for anything and it's not forgotten from somewhere else.

Reply:
To be honest, I didn't dig deep into details,
for some reason `WebIdentityTokenCredentialsProvider` without region specified in the provider was failing with this error:

```
Getting object test.txt from bucket neon-dev-extensions-eu-west-1 Error: DispatchFailure(DispatchFailure { source: ConnectorError { kind: Other(None), source: ProviderError(ProviderError { source: DispatchFailure(DispatchFailure { source: ConnectorError { kind: Other(None), source: ResolveEndpointError { message: "Invalid Configuration: Missing Region", source: None }, connection: Unknown } }) }), connection: Unknown } })
```